tests/core/string/test_fuzzy_search.h

```diff
@@ -0,0 +1,157 @@
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_FUZZY_SEARCH_H
+#define TEST_FUZZY_SEARCH_H
+
+#include "core/string/fuzzy_search.h"
+#include "tests/test_macros.h"
+#include <chrono>
+#include <iostream>
+
+namespace TestFuzzySearch {
+
+struct FuzzySearchTestCase {
+	String query;
+	String expected;
+};
+
+struct FuzzySearchTestOutcome {
+	String top_result;
+	int result_count;
+};
+
+struct FuzzySearchBenchmarkResult {
+	double average_ms;
+	double std_dev_ms;
+	FuzzySearchTestOutcome outcome;
+};
+
+// Ideally each of these test queries should represent a different aspect, and potentially bottleneck, of the search process.
+const FuzzySearchTestCase test_cases[] = {
+	// Short query, many matches, few adjacent characters
+	{ "///gd", "./menu/hud/hud.gd" },
+	// Filename match with typo
+	{ "sm.png", "./entity/blood_sword/sam.png" },
+	// Multipart filename word matches
+	{ "ham ", "./entity/game_trap/ha_missed_me.wav" },
+	// Single word token matches
+	{ "push background", "./entity/background_zone1/background/push.png" },
+	// Long token matches
+	{ "background_freighter background png", "./entity/background_freighter/background/background.png" },
+	// Many matches, many short tokens
+	{ "menu menu characters wav", "./menu/menu/characters/smoker/0.wav" },
+	// Maximize total matches
+	{ "entity gd", "./entity/entity_man.gd" }
+};
+
+double calculate_mean(const Vector<double> &p_numbers) {
+	double sum = 0.0;
+	for (double num : p_numbers) {
+		sum += num;
+	}
+	return sum / static_cast<double>(p_numbers.size());
+}
+
+double calculate_std_dev(const Vector<double> &p_numbers) {
+	double mean = calculate_mean(p_numbers);
+	double variance = 0.0;
+	for (double num : p_numbers) {
+		variance += (num - mean) * (num - mean);
+	}
+	variance /= static_cast<double>(p_numbers.size());
+	return std::sqrt(variance);
+}
+
+auto load_test_data(int p_repeat = 1) {
+	// This file has 1k entries so p_repeat can be used to benchmark in multiples of 1k
+	Ref<FileAccess> fp = FileAccess::open(TestUtils::get_data_path("fuzzy_search/project_dir_tree.txt"), FileAccess::READ);
+	REQUIRE(!fp.is_null());
+	auto lines = fp->get_as_utf8_string().split("\n");
+	Vector<String> all_lines;
+	while (p_repeat-- > 0) {
+		all_lines.append_array(lines);
+	}
+	CHECK(lines.size() > 0);
+	return all_lines;
+}
+
+FuzzySearchTestOutcome get_top_result_and_count(String &p_query, Vector<String> &p_lines, int p_max_results = 100) {
+	FuzzySearch search;
+	search.set_query(p_query);
+	search.max_results = p_max_results;
+	Vector<FuzzySearchResult> results;
+	search.search_all(p_lines, results);
+	return { results.size() > 0 ? results[0].target : "<no result>", (int)results.size() };
+}
+
+FuzzySearchBenchmarkResult bench(String p_query, Vector<String> p_targets) {
```

Comment:
Unused, please remove, we don't put benchmarks in the tests anyway

Reply:
@AThousandShips where should benchmarks go? It's important to retain the ability to measure search speed for future enhancements. I wouldn’t want to lose this.

Reply:
We have repository for benchmarks: https://github.com/godotengine/godot-benchmarks
I'm unsure about the process of adding one, but I think you could extract this code into patch, then attach it in a new issue on the repo.